shelley/chain-and-ledger/shelley-spec-ledger-test/src/Test/Shelley/Spec/Ledger/Serialisation/Generators.hs

```diff
@@ -284,508 +75,56 @@ instance
-                <*> (listOf (sizedMetaDatum (n -1)))
-            ),
-      MD.List <$> resize maxMetaDatumListLens (listOf (sizedMetaDatum (n -1))),
-      MD.I <$> arbitrary,
-      MD.B <$> arbitrary,
-      MD.S <$> (T.pack <$> arbitrary)
-    ]
-
-instance Arbitrary MD.MetaDatum where
-  arbitrary = sizedMetaDatum maxMetaDatumDepth
-
-instance Arbitrary MD.MetaData where
-  arbitrary = MD.MetaData <$> arbitrary
-
-maxTxWits :: Int
-maxTxWits = 5
-
 instance
-  ( ShelleyTest era,
-    Mock (Crypto era),
-    Arbitrary (WitnessSet era),
-    Arbitrary (Core.Value era)
+  ( Mock c,
+    PraosCrypto c,
+    Arbitrary (WitnessSet (ShelleyEra c)),
+    Arbitrary (Core.Value (ShelleyEra c))
   ) =>
-  Arbitrary (Tx era)
+  Arbitrary (Tx (ShelleyEra c))
   where
-  -- Our arbitrary instance constructs things using the pattern in order to have
-  -- the correct serialised bytes.
-  arbitrary =
-    Tx
-      <$> arbitrary
-      <*> (resize maxTxWits arbitrary)
-      <*> arbitrary
-
-instance Era era => Arbitrary (TxId era) where
-  arbitrary = TxId <$> genHash
-
-instance Era era => Arbitrary (TxIn era) where
-  arbitrary =
-    TxIn
-      <$> (TxId <$> genHash)
-      <*> arbitrary
-
-instance
-  (ShelleyTest era, Mock (Crypto era), Arbitrary (Core.Value era)) =>
-  Arbitrary (TxOut era)
-  where
-  arbitrary = TxOut <$> arbitrary <*> arbitrary
-
-instance Arbitrary Nonce where
-  arbitrary =
-    oneof
-      [ return NeutralNonce,
-        mkNonceFromNumber <$> choose (1, 123 :: Word64)
-      ]
-
-instance Arbitrary UnitInterval where
-  arbitrary = fromJust . mkUnitInterval . (% 100) <$> choose (1, 99)
+  arbitrary = genTx
 
 instance
-  (CC.Crypto crypto) =>
-  Arbitrary (KeyHash a crypto)
+  ( Mock c,
+    PraosCrypto c,
+    Arbitrary (STS.PredicateFailure (UTXO (ShelleyEra c)))
+  ) =>
+  Arbitrary (STS.UtxowPredicateFailure (ShelleyEra c))
   where
-  arbitrary = KeyHash <$> genHash
-
-instance Era era => Arbitrary (WitHashes era) where
-  arbitrary = genericArbitraryU
-
-instance Arbitrary MIRPot where
-  arbitrary = genericArbitraryU
-
-instance Arbitrary Natural where
-  arbitrary = fromInteger <$> choose (0, 1000)
-
-instance Arbitrary STS.VotingPeriod where
-  arbitrary = genericArbitraryU
-  shrink = genericShrink
-
-instance Era era => Arbitrary (STS.PpupPredicateFailure era) where
   arbitrary = genericArbitraryU
   shrink = genericShrink
 
 instance
-  ( ShelleyTest era,
-    Mock (Crypto era),
-    Arbitrary (Core.Value era),
-    Arbitrary (Delta (Core.Value era))
+  ( Mock c,
+    PraosCrypto c,
+    PredicateFailure (UTXO (ShelleyEra c)) ~ STS.UtxoPredicateFailure (ShelleyEra c)
   ) =>
-  Arbitrary (STS.UtxoPredicateFailure era)
+  Arbitrary (STS.UtxoPredicateFailure (ShelleyEra c))
   where
   arbitrary = genericArbitraryU
+
   -- we don't have a shrinker for Value, so we do not shrink this
   -- predicate failure, as its constructor contains Value
   shrink pf = [pf]
```

Comment:
BTW, if you can't shrink, return `[]`. I think the current code might cause an infinite shrink.